### DISCUSSION TITLE
Complain about removing context.read on Riverpod 1.0.0

### DISCUSSION THREAD

**Discussion by heda7** [rrousselGit/riverpod#982] [Q&A] [answered]
I used to pass context to logic side on my project so I can call notifiers methods inside them using the extension method "context.read"

That made a good separation between UI and logic code, But now this extension method is removed on Riverpod 1.0.0 and I can't migrate from 0.14.0+3 

Can anyone help?

**Comment by heda7** [comment]
I found the following package on pub.dev 

https://pub.dev/packages/riverpod_context

But I really want to use the original package to get all the good updates ^_^

**Reply by rrousselGit** [reply]
I'd recommend against using that. It's not a stable solution

**Comment by rrousselGit** [comment, marked as answer]
Why can't you migrate?

You should be able to switch to ConsumerWidget and use WidgetRef.read

**Reply by heda7** [reply]
Because in this case I would pass the WidgetRef to the logic functions and you advise against that on the docs.

I read that i should not use the WidgetRef outside the ConsumerWidget.

**Reply by rrousselGit** [reply]
The doc is probably misleading then.

It is about not passing "ref" to other widgets, but passing the ref to classic functions is fine
